presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractFixedWidthBlock.java

```diff
@@ -113,13 +114,14 @@ public boolean bytesEqual(int position, int offset, Slice otherSlice, int otherO
     }
 
     @Override
-    public int hash(int position, int offset, int length)
+    public long hash(int position, int offset, int length)
     {
         checkReadablePosition(position);
         if (isNull(position)) {
             return 0;
         }
-        return getRawSlice().hashCode(valueOffset(position) + offset, length);
+
+        return XxHash64.hash(getRawSlice(), valueOffset(position) + offset, length); // TODO: this can be patched into airlift
```

Comment:
This TODO is invalid, I forgot to remove it - as you can see it already is available in airlift - I just missed it on first iteration ;-)